### PR TITLE
OHRM5X-2088: Refactor Admin FE unique validation to BE validation

### PR DESCRIPTION
**Pull request by Super-Chama**

### CODE REVIEW

File: src/plugins/orangehrmAdminPlugin/Api/ValidationUniqueAPI.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\ResourceEndpoint;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+
+class ValidationUniqueAPI extends Endpoint implements ResourceEndpoint
```
[the remark this answers is not in the document]
Comment (Super-Chama) [reply src/plugins/orangehrmAdminPlugin/Api/ValidationUniqueAPI.php:31]: done in https://github.com/Super-Chama/orangehrm/pull/74

File: src/plugins/orangehrmAdminPlugin/Api/ValidationUniqueAPI.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        $paramsRules = new ParamRuleCollection();
+        $paramsRules->addExcludedParamKey(self::PARAMETER_VALUE);
+        $paramsRules->addExcludedParamKey(self::PARAMETER_ENTITY_Id);
+        $paramsRules->addExcludedParamKey(self::PARAMETER_ENTITY_NAME);
+        $paramsRules->addExcludedParamKey(self::PARAMETER_ATTRIBUTE_NAME);
+        $paramsRules->addExcludedParamKey(self::PARAMETER_MATCH_BY_FIELD);
+        $paramsRules->addExcludedParamKey(self::PARAMETER_MATCH_BY_VALUE);
```
Comment (devishke-orange) [inline src/plugins/orangehrmAdminPlugin/Api/ValidationUniqueAPI.php:80]: should add length validation

**Comment by devishke-orange** [comment]
add unit and integration test

File: src/client/src/orangehrmLeavePlugin/pages/configure/holiday/SaveHoliday.vue [hunk 3]
```diff
@@ -145,43 +152,16 @@ export default {
       holiday: {...holidayModel},
       rules: {
         name: [required, shouldNotExceedCharLength(200)],
-        date: [required, validDateFormat(this.userDateFormat)],
+        date: [
+          required,
+          validDateFormat(this.userDateFormat),
+          this.holidayDateUniqueValidation,
+        ],
```
Comment (devishke-orange) [inline src/client/src/orangehrmLeavePlugin/pages/configure/holiday/SaveHoliday.vue:159]: not working for checking recurring holidays

File: src/plugins/orangehrmAdminPlugin/Api/ValidationUniqueAPI.php [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    public const PARAMETER_ENTITY_Id = 'entityId';
+    public const PARAMETER_ENTITY_NAME = 'entityName';
+    public const PARAMETER_ATTRIBUTE_NAME = 'attributeName';
+    public const PARAMETER_MATCH_BY_FIELD = 'matchByField';
+    public const PARAMETER_MATCH_BY_VALUE = 'matchByValue';
+    public const PARAMETER_IS_UNIQUE_RECORD = 'valid';
+
+    private ?ValidationUniqueDao $validationUniqueDao = null;
+
+    public function getOne(): EndpointResult
```
Comment (devishke-orange) [inline src/plugins/orangehrmAdminPlugin/Api/ValidationUniqueAPI.php:43]: missing php annotation and api doc

File: src/plugins/orangehrmAdminPlugin/Api/ValidationUniqueAPI.php (hunk 2 above)
Comment (devishke-orange) [reply src/plugins/orangehrmAdminPlugin/Api/ValidationUniqueAPI.php]: added

File: src/client/src/orangehrmLeavePlugin/pages/configure/holiday/SaveHoliday.vue (hunk 3 above)
Comment (devishke-orange) [reply src/client/src/orangehrmLeavePlugin/pages/configure/holiday/SaveHoliday.vue]: reverted this

File: src/plugins/orangehrmAdminPlugin/Api/ValidationUniqueAPI.php (hunk 4 above)
Comment (devishke-orange) [reply src/plugins/orangehrmAdminPlugin/Api/ValidationUniqueAPI.php:104]: added

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -62,7 +62,7 @@ jobs:
             -e LDAP_CONFIG_ADMIN_PASSWORD=admin \
             -e LDAP_SKIP_DEFAULT_TREE=yes \
             -v $PWD/ldifs:/ldifs \
-            bitnami/openldap:latest
+            bitnami/openldap:2.6.4
```
Comment (devishke-orange) [inline .github/workflows/test.yml:65]: temporary change due to test failure on latest LDAP container

**Comment by Super-Chama** [comment]
Closing due to https://github.com/orangehrm/orangehrm/pull/1723

**Pull request closed without merging**